### PR TITLE
fix(extension): [lw-10539] shared wallet settings section

### PR DESCRIPTION
**Pull request by vetalcore**
# Checklist

- [x] [JIRA - LW-10539](https://input-output.atlassian.net/browse/LW-10539)
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Attach screenshots here if implementation involves some UI changes

### CODE REVIEW

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const isValidSharedWalletScript = (script: Cardano.NativeScript): boolean => {
+  switch (script.kind) {
+    case Cardano.NativeScriptKind.RequireAllOf:
+    case Cardano.NativeScriptKind.RequireAnyOf:
+    case Cardano.NativeScriptKind.RequireNOf:
+      return script.scripts.every((nativeScript) => nativeScript.kind === Cardano.NativeScriptKind.RequireSignature);
+    default:
+      return false;
+  }
+};
```
Comment (vetalcore) [inline apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts:15]: this one exists but not exported from `cardano-js-sdk`*
could create a pr to export it if required?
@szymonmaslowski wdyt?

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  return walletAddresses?.some(
+    (addr) =>
+      isScriptAddress(addr) &&
+      Cardano.isNativeScript(addr.scripts.payment) &&
+      isValidSharedWalletScript(addr.scripts.payment) &&
+      Cardano.isNativeScript(addr.scripts.stake) &&
+      isValidSharedWalletScript(addr.scripts.stake)
+  );
+};
```
Comment (vetalcore) [inline apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts:28]: not sure this is a proper way to check if current wallet is shared, i've created shared wallet but there are no scripts found in the addresses observable
@szymonmaslowski could you please suggest?

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { Cardano } from '@cardano-sdk/core';
```
Comment (vetalcore) [inline apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts:1]: should it live in the  `share-wallet` package?
also there are `isHardwareWallet` and `isInMemoryWallet` exported  from the `walletInfoSlice`,  would it be a proper place for `isSaredWallet` as well?

Comment (szymonmaslowski) [reply apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts:1]: I think it is a good idea to have it in the `walletInfoSlice` along with `isHardwareWallet` and `isInMemoryWallet` :+1:

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts (hunk 1 above)
Comment (szymonmaslowski) [reply apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts:15]: Yeah, why not! I guess exporting this utility from sdk will be pretty quick and then we could update the sdk package version in this PR. Thanks for spotting that!

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts (hunk 2 above)
Comment (szymonmaslowski) [reply apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts:28]: Try the following way: (I didn't try it, just scratched this code)
```
  const { walletId } = useObservable(walletManager.activeWalletId$);
  const wallets = useObservable(walletRepository.wallets$);
  const { paymentScript, stakingScript } = wallets.find((w) => w.walletId === walletId)
```

**Review by szymonmaslowski — COMMENTED** [review]
GJ!

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts (hunk 1 above)
Comment (vetalcore) [reply apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts]: resolved here https://github.com/input-output-hk/lace/pull/1231/commits/ddcf042e918f20fe4aa113ae43428156e84d0301

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts (hunk 2 above)
Comment (vetalcore) [reply apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts]: done here https://github.com/input-output-hk/lace/pull/1231/commits/ddcf042e918f20fe4aa113ae43428156e84d0301

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts (hunk 3 above)
Comment (vetalcore) [reply apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts]: done here https://github.com/input-output-hk/lace/pull/1231/commits/ddcf042e918f20fe4aa113ae43428156e84d0301

File: apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts (hunk 1 above)
Comment (vetalcore) [reply apps/browser-extension-wallet/src/views/browser-view/features/settings/components/utils/isSharedWallet.ts]: cardano js sdk is bumped now

File: apps/browser-extension-wallet/src/utils/is-shared-wallet.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { Cardano } from '@cardano-sdk/core';
+import { isValidSharedWalletScript } from '@cardano-sdk/wallet';
+import { WalletType } from '@cardano-sdk/web-extension';
+import { Wallet } from '@lace/cardano';
+
+export const isSharedWallet = (wallet?: Wallet.CardanoWallet): boolean => {
+  if (wallet?.source.wallet.type !== WalletType.Script) return false;
```
Comment (vetalcore) [inline apps/browser-extension-wallet/src/utils/is-shared-wallet.ts:7]: `wallet?.source.wallet.type` gives `inMemory` for me
@szymonmaslowski are we actually creating a shared wallet?
if no - how are we going to pass it to the sdet?

File: packages/shared-wallets/src/add-shared-wallet/main-page-flow/SharedWalletGetStarted/SharedWalletGetStarted.tsx [hunk 5]
```diff
@@ -21,12 +21,12 @@ export type SharedWalletGetStartedSharedProps = {
 type SharedWalletGetStartedProps = SharedWalletGetStartedSharedProps &
   (
     | {
-        createAndImportOptionsDisabled: true;
+        createAndImportOptionsDisabled: boolean;
```
Comment (szymonmaslowski) [inline packages/shared-wallets/src/add-shared-wallet/main-page-flow/SharedWalletGetStarted/SharedWalletGetStarted.tsx:24]: I guess you changed it because there was an build error of the shared-wallet package, right? That is my mistake, somehow it sneaked into the main. The error is fixed [here](https://github.com/input-output-hk/lace/pull/1232/files#diff-6401b8028220f99b00f8628a06a58c3488761a30c90bc29b847d498d46fb2459R9), let's revert the changes in this file.

**Review by szymonmaslowski — COMMENTED** [review]
Great Job! This one anyway will need to wait for the changes regarding shared-wallet instance creation. Not approving yet just because of changes in the `SharedWalletGetStarted.tsx`, they should be ready to be removed soon.

**Review by danielmain — APPROVED** [review]
Great job @vetalcore

File: apps/browser-extension-wallet/src/utils/is-shared-wallet.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { Cardano } from '@cardano-sdk/core';
+import { isValidSharedWalletScript } from '@cardano-sdk/wallet';
+import { WalletType } from '@cardano-sdk/web-extension';
+import { Wallet } from '@lace/cardano';
+
+export const isSharedWallet = (wallet?: Wallet.CardanoWallet): boolean => {
```
Comment (szymonmaslowski) [inline apps/browser-extension-wallet/src/utils/is-shared-wallet.ts:6]: I would move it to the shared-wallets space. Let's do it after merging this PR :wink:

**Review by szymonmaslowski — APPROVED** [review]
GJ! I left one suggestion for later. Please also take a look on sonar cube report.

File: packages/shared-wallets/src/add-shared-wallet/main-page-flow/SharedWalletGetStarted/SharedWalletGetStarted.tsx (hunk 5 above)
Comment (vetalcore) [reply packages/shared-wallets/src/add-shared-wallet/main-page-flow/SharedWalletGetStarted/SharedWalletGetStarted.tsx]: done here 3a39e8a0a1d97d8e588f5d7ed0a245578a2a8219
